src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -467,16 +468,22 @@ static TypeSymbol RemoveImplicitNull(TypeSymbol type, bool typeWasPreserved)
 
             var missingRequiredProperties = targetType.Properties.Values
                 .Where(p => p.Flags.HasFlag(TypePropertyFlags.Required) && !namedPropertyMap.ContainsKey(p.Name))
-                .Select(p => p.Name)
-                .OrderBy(p => p);
+                .ToList();
+                
 
-            if (missingRequiredProperties.Any())
+            if (missingRequiredProperties.Count > 0)
```

Comment:
>Count > 0 [](http://example.com/codeflow?start=42&length=9)

It's fine to change it, but `Any()` has a special case for `ICollection` to use `Count` instead of scanning through the enumerable.

Reply:
ok, I remember that some time ago there wasn't this special casing and `Count > 0` was faster in some benchmarks.